From 9858001e70d3a4dad13386521f12f072709eaef9 Mon Sep 17 00:00:00 2001
Message-Id: <9858001e70d3a4dad13386521f12f072709eaef9.1439192123.git.chang-joon.lee@intel.com>
From: vandita kulkarni <vandita.kulkarni@intel.com>
Date: Tue, 28 Jul 2015 23:15:40 +0530
Subject: [PATCH] SQUASHME! [VPG]: drm/i915: Fix the DL value during vp

This patch fixes the dl value and programs it SV
recommended value during video playback on dual display,
ie single plane sprite case, which was missed out in
the earlier patch.

SQUASHME! this patch needs to be squashed with
    	Change-Id: I8f3db5bc0f2a1688521304aad1ab84753c0b671c
	Author: vandita kulkarni <vandita.kulkarni@intel.com>
	Date:   Fri May 8 19:44:51 2015 +0530
		REVERTME: [VPG]: drm/i915: update dl in multipipe
		ISSUE: IMINAN-36379

Change-Id: I217a039f2640cf38466e8bd5ea46ed11f14e9e46
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-42160
Signed-off-by: vandita kulkarni <vandita.kulkarni@intel.com>
## Don't forget to add a JIRA URL to the Tracked-On metadata!
## https://wiki.ith.intel.com/display/CACTUS/Patch+requirements+check
## url should be in the form:
## Tracked-On: https://jira01.devtools.intel.com/browse/<PROJ>-<NUMB>
## Several Tracked-On lines are allowed
## you can fully prefill your Tracked-On commits using environment variable:
## export CURRENT_TRACKED_ON=https://jira01.devtools.intel.com/browse/<PROJ>-<NUMB>
##
---
 drivers/gpu/drm/i915/intel_sprite.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_sprite.c b/drivers/gpu/drm/i915/intel_sprite.c
index dc5d0c4..dac30b8 100644
--- a/drivers/gpu/drm/i915/intel_sprite.c
+++ b/drivers/gpu/drm/i915/intel_sprite.c
@@ -413,6 +413,7 @@ vlv_update_plane(struct drm_plane *dplane, struct drm_crtc *crtc,
 	struct intel_crtc *intel_crtc = to_intel_crtc(crtc);
 	int pipe = intel_plane->pipe;
 	int plane = intel_plane->plane;
+	int pipe_stat = VLV_PIPE_STATS(dev_priv->pipe_plane_stat);
 	u32 sprctl;
 	bool rotate = false;
 	bool alpha_changed = false;
@@ -652,6 +653,19 @@ vlv_update_plane(struct drm_plane *dplane, struct drm_crtc *crtc,
 	vlv_calculate_ddl(crtc, pixel_size, &sp_prec_multi, &sprite_ddl);
 	sprite_ddl = (sp_prec_multi | sprite_ddl) << shift;
 
+	/*
+	 * The current Dl formula doesnt consider multipipe
+	 * cases, Use this value suggested by sv till the
+	 * actual formula gets used, same applies for all
+	 * hdmi cases. Since secondary display comes on PIPEC
+	 * we are checking for pipe C, pipe_stat variable
+	 * tells us the number of pipes enabled.
+	 */
+	if (IS_CHERRYVIEW(dev))
+		if (!single_pipe_enabled(pipe_stat) ||
+				(pipe_stat & PIPE_ENABLE(PIPE_C)))
+			sprite_ddl = DDL_MULTI_PIPE_CHV << shift;
+
 	if (intel_plane->plane) {
 		intel_crtc->reg_ddl.spriteb_ddl = sprite_ddl;
 		intel_crtc->reg_ddl.spriteb_ddl_mask = mask;
-- 
1.9.1

